### PR TITLE
[DF] Test against reproducer of #7584

### PR DESCRIPTION
**Pull request by vepadulano**
Sibling PR of https://github.com/root-project/root/pull/8605

Tests against reproducer of root-project/root#7584

### CODE REVIEW

File: python/distrdf/test_friend_trees.py [hunk 1]
```diff
@@ -107,6 +108,55 @@ def test_friend_tree_histo(self):
         os.remove("treeparent.root")
         os.remove("treefriend.root")
 
+    def test_friend_trees_7584(self):
```
Comment (eguiraud) [inline python/distrdf/test_friend_trees.py:111]: 7584 does not really add much :smile: 
```suggestion
    def test_friend_trees(self):
```

File: python/distrdf/test_friend_trees.py [hunk 2]
```diff
@@ -107,6 +108,55 @@ def test_friend_tree_histo(self):
...
+        """Test against the reproducer of issue #7584"""
```
Comment (eguiraud) [inline python/distrdf/test_friend_trees.py:112]:
```suggestion
        """Test against the reproducer of https://github.com/root-project/root/issues/7584"""
```

**Review by eguiraud — APPROVED** [review]
Assuming you tested that this actually fails without https://github.com/root-project/root/pull/8605 :smile:

**Comment by vepadulano** [comment]
Indeed it fails without the linked PR. 
```
vpadulan@fedorathinkpad-T550 [~]: root --version
ROOT Version: 6.24/00
Built for linuxx8664gcc on Apr 22 2021, 10:24:00
From tags/v6-24-00@v6-24-00
vpadulan@fedorathinkpad-T550 [~]: python reproducer_7584.py
...
ERROR: test_friend_trees_7584 (__main__.SparkFriendTreesTest)
Test against the reproducer of issue #7584
----------------------------------------------------------------------
Traceback (most recent call last):
  File "reproducer_7584.py", line 147, in test_friend_trees_7584
    self.assertEqual(h_parent.GetEntries(), 10000)
  File "/home/vpadulan/Programs/rootproject/rootinstall/v6-24-00/lib/DistRDF/Proxy.py", line 136, in _call_action_result
    return getattr(self.GetValue(), self._cur_attr)(*args, **kwargs)
  File "/home/vpadulan/Programs/rootproject/rootinstall/v6-24-00/lib/DistRDF/Proxy.py", line 127, in GetValue
    headnode.backend.execute(generator)
  File "/home/vpadulan/Programs/rootproject/rootinstall/v6-24-00/lib/DistRDF/Backends/Base.py", line 126, in execute
    ranges = headnode.build_ranges()
  File "/home/vpadulan/Programs/rootproject/rootinstall/v6-24-00/lib/DistRDF/Node.py", line 632, in build_ranges
    return RangesBuilder(self).build_ranges()
  File "/home/vpadulan/Programs/rootproject/rootinstall/v6-24-00/lib/DistRDF/Node.py", line 591, in build_ranges
    return self._get_clustered_ranges(treename, inputfiles, friendinfo)
  File "/home/vpadulan/Programs/rootproject/rootinstall/v6-24-00/lib/DistRDF/Node.py", line 459, in _get_clustered_ranges
    clustersinfiles = self.get_clusters(treename, filelist)
  File "/home/vpadulan/Programs/rootproject/rootinstall/v6-24-00/lib/DistRDF/Node.py", line 367, in get_clusters
    entries = t.GetEntriesFast()
AttributeError: 'TObject' object has no attribute 'GetEntriesFast'
```

**Comment by eguiraud** [comment]
is `TObject has no attribute GetEntriesFast` the expected failure mode? I thought the issue was `runtime_error: Unknown column: myfriend.rnd`.

**Comment by vepadulano** [comment]
There are multiple points of failure with this test before the linked PR. In this particular case the reason is that in `Node.RangesBuilder.get_clusters` we want retrieve a tree named `randomNumbers` from a file `friendtrees_spark.root`. And if we print those variables we get
```
filename = friendtrees_spark.root
treename = 
```
This happens because the user created a TChain with no name and then added the trees by issuing "filename/treename".

**Comment by vepadulano** [comment]
The error `runtime_error: Unknown column: myfriend.rnd` appears if in the reproducer we add the names of the trees in the `TChain` constructors. At that point the issue is that there was no support for aliases of friends before the linked PR

File: python/distrdf/test_friend_trees.py (hunk 1 above)
Comment (vepadulano) [reply python/distrdf/test_friend_trees.py]: fixed in rebased commit

File: python/distrdf/test_friend_trees.py (hunk 2 above)
Comment (vepadulano) [reply python/distrdf/test_friend_trees.py]: fixed in rebased commit

**Comment by vepadulano** [comment]
@phsft-bot build

**Pull request merged**